apps/shared/y4m.c

```diff
@@ -132,6 +137,23 @@ static avifBool y4mColorSpaceParse(const char * formatString, struct y4mFrameIte
     return AVIF_FALSE;
 }
 
+// Returns an unsigned integer value parsed from [start:end[.
+// Returns -1 in case of failure.
+int y4mReadUnsignedInt(const char * start, const char * end)
+{
+    if (start >= end) {
+        return -1;
+    }
+    int64_t value = 0;
+    while (start != end && *start >= '0' && *start <= '9') {
+        value = value * 10 + (*(start++) - '0');
+        if (value > INT_MAX) {
+            return -1;
+        }
+    }
+    return (int)value;
```

Comment:
IMPORTANT: Return -1 if the input contains no decimal digits.

This is harmless because the caller also treats 0 as an error.